### DISCUSSION TITLE
Calculating average steel element temperature in FDS? [feature request]

### DISCUSSION THREAD

**Discussion by Er9y714** [firemodels/fds#12224] [Ideas]
Ozone is a zone model which also calculates protected and unprotected steel element temperature based on Eurocode methods. Basically gas temperature and radiative heat flux is taken into account and applied to the element with a certain shape factor. Since these inputs are present in FDS as DEVCs already, would it be interesting to add such a feature of predicting average steel element temperature by adding a few more options (Steel profile type or shape factor, number of side the steel element is exposed etc.)? 

It would be also interesting to discuss the concerns of the method. After all, there are many assumptions and simplifications in this Eurocode method. 

Thanks,

https://dan.ct.upt.ro/fire/files/OZone_V2.pdf

https://research.bauforumstahl.de/nl/fire-safety-1/locafi-temperature-assessment-of-a-vertical-member-subjected-to-localised-fire-10/

**Comment by drjfloyd** [comment]
I think these outputs are not something we would look to incorporate into FDS. Over time these kinds of outputs tend to make code maintenance more difficult. Also this is tied to a specific code and codes evolve and change over time and we which also causes a maintenance issue sas we do not keep track of all the code requirements worldwide.  This looks like something that would be better done in post-processing or via use of CTRL and DEVC funcitons.

**Comment by Er9y714** [comment]
Totally makes sense. Thank you for sharing this.
